modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj

```diff
@@ -553,80 +553,64 @@
     (cond->> ((get-method sql.qp/->honeysql [:sql :relative-datetime]) driver clause)
       t (->temporal-type t))))
 
+(defn- datetime-diff-check-args
+  "Validates the types of the datetime args to a `datetime-diff` clause. Bigquery's type handling
+   differs from other drivers, which means we can't use [[sql.qp/datetime-diff-check-args]]."
+  [x y]
+  (doseq [arg [x y]
+          :let [db-type (some-> (temporal-type arg) name)]
+          :when (and db-type (not (re-find #"^(?i)(timestamp|date)" db-type)))]
+    (throw (ex-info (tru "datetimeDiff only allows datetime, timestamp, or date types. Found {0}"
+                         (pr-str db-type))
+                    {:found db-type
+                     :type  qp.error-type/invalid-query}))))
```

Comment:
This is exactly the same as `sql.qp/datetime-diff-check-args` except we're checking the `temporal-type`, not the `database-type` of the arg.

Reply:
instead of writing this in a comment on the PR, you should write this in a comment in the code itself.

Save the PR comments for things that can't be commented in the code itself, like moving something around or deleting code

Reply:
Good point, updated https://github.com/metabase/metabase/commit/45e0a6b1e98b48b02563daf68a2b24ed79bc12cf